src/app/components/kudos-user-history/kudos-user-history.service.js

```diff
@@ -0,0 +1,161 @@
+(function () {
+    "use strict";
+    angular.module('myApp.components.userHistory')
+        .factory("UserHistoryService", UserHistoryService);
+
+    UserHistoryService.$inject = [
+        "$httpParamSerializer",
+        "Transaction"
+    ];
+
+    function UserHistoryService($httpParamSerializer, Transaction) {
+        var transactionsAll = {},
+            transactionsReceived = {},
+            transactionsGave = {},
+            transactionsChallenges = {};
+        
+        var users = [],
+            index;
+
+        var service = {
+            getAllTransactions : getAllTransactions,
+            getReceivedTransactions : getReceivedTransactions,
+            getGaveTransactions : getGaveTransactions,
+            getChallengesTransactions : getChallengesTransactions,
+            getAll : getAllUserTransactions,
+            setAll : setAllUserTransactions,
+            getReceived : getReceivedUserTransactions,
+            setReceived : setReceivedUserTransactions,
+            getGave : getGaveUserTransactions,
+            setGave : setGaveUserTransactions,
+            getChallenges : getChallengesUserTransactions,
+            setChallenges : setChallengesUserTransactions
+        };
+        return service;
+
+        function getAllTransactions(requestData) {
+            if (requestData){
+                requestData = $httpParamSerializer({
```

Comment:
Do you really need `$httpParamSerializer` here? I was under the influence, that` as long as you provide the correct `Content-Type` header, everything is serialized automatically?